internal/storage/bucket_handle_test.go

```diff
@@ -256,3 +257,94 @@ func (t *BucketHandleTest) TestCreateObjectMethodWhenGivenGenerationObjectNotExi
 	AssertEq(nil, obj)
 	AssertTrue(strings.Contains(err.Error(), "Error 412: Precondition failed"))
 }
+
+func (t *BucketHandleTest) TestGetProjectValueWhenGcloudProjectionIsNoAcl() {
+	proj := getProjectionValue(gcs.Projection(storage.ProjectionNoACL))
+
+	AssertEq(storage.Projection(1), proj)
+}
+
+func (t *BucketHandleTest) TestGetProjectValueWhenGcloudProjectionIsFull() {
+	proj := getProjectionValue(gcs.Projection(storage.ProjectionFull))
+
+	AssertEq(storage.Projection(2), proj)
+}
+
+func (t *BucketHandleTest) TestGetProjectValueWhenGcloudProjectionIsDefault() {
+	proj := getProjectionValue(gcs.Projection(storage.ProjectionDefault))
+
+	AssertEq(storage.Projection(1), proj)
+}
+
+func (t *BucketHandleTest) TestListObjectMethodWithPrefixObjectExist() {
+	obj, err := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "gcsfuse/",
+			Delimiter:                "/",
+			IncludeTrailingDelimiter: true,
+			ContinuationToken:        "ContinuationToken",
+			MaxResults:               7,
+			ProjectionVal:            0,
+		})
+
+	AssertEq(nil, err)
+	AssertEq(TestObjectRootFolderName, obj.Objects[0].Name)
+	AssertEq(TestObjectSubRootFolderName, obj.Objects[1].Name)
+	AssertEq(TestObjectName, obj.Objects[2].Name)
+	AssertEq(TestObjectGeneration, obj.Objects[0].Generation)
+	AssertEq(TestObjectSubRootFolderName, obj.CollapsedRuns[0])
+}
+
+func (t *BucketHandleTest) TestListObjectMethodWithPrefixObjectDoesNotExist() {
+	obj, err := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "PrefixObjectDoesNotExist",
+			Delimiter:                "/",
+			IncludeTrailingDelimiter: true,
+			ContinuationToken:        "ContinuationToken",
+			MaxResults:               7,
+			ProjectionVal:            0,
+		})
+
+	AssertEq(nil, err)
+	AssertEq(nil, obj.Objects)
+	AssertEq(nil, obj.CollapsedRuns)
+}
+
+func (t *BucketHandleTest) TestListObjectMethodWithIncludeTrailingDelimiterFalse() {
+	obj, err := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "gcsfuse/",
+			Delimiter:                "/",
+			IncludeTrailingDelimiter: false,
+			ContinuationToken:        "ContinuationToken",
+			MaxResults:               7,
+			ProjectionVal:            0,
+		})
+
+	AssertEq(nil, err)
+	AssertEq(TestObjectRootFolderName, obj.Objects[0].Name)
```

Comment:
also check that no other objects should be present in the response by adding an assert on the size. same for all tests.

Reply:
fixed